File: app/src/main/java/com/sofrosyn/a6gnlsolarcalculator/logic/Calculator.java
package com.sofrosyn.a6gnlsolarcalculator.logic;

import java.text.DecimalFormat;

public  class Calculator {

  private static double inverterRatingKva;
  private static int backUpTime;
  private static int timeToCharge;
  private static int minRatingChargeController;
  private static int noSolarPanels;

  private static final double LOADFACTOR = 0.8;
  private static final int AMPHOURBATTERY = 200;
  private static final int HOURS = 8;



  public static double loadInKva(double load ){

      inverterRatingKva = (load / LOADFACTOR)/ 1000;
        return inverterRatingKva;
      }


      public static float batteryBackupTime(float load, float inverterVoltage,float batteryAmp){
         float cal = load / inverterVoltage;
          return batteryAmp / cal;

      }

      public static float timeToCharge(float batteryAmps,float inverterVoltage){
      return batteryAmps / inverterVoltage;
      }


      public static int minCurrentRatingController(int panelPowerRating,int panelVoltage, int noSolarPanel){

      minRatingChargeController  = panelPowerRating * noSolarPanel;
      return minRatingChargeController / panelVoltage;
  }


      public static int totalPanels(int inverterVoltage, int panelRating){

    int a = AMPHOURBATTERY * inverterVoltage;
    int b = HOURS  * panelRating;
          noSolarPanels = a / b;
      return noSolarPanels;
      }


      public static int ampsConverter(int batterNumber,int divisor,int ampHourBattery){

      int base = batterNumber / divisor;
      return base * ampHourBattery;


      }

    public static double KvaMargin(double load ){


        return load + 0.3*load;
    }



    public static String formatDecimal(double value){
          DecimalFormat decimalFormat = new DecimalFormat("#.##");
            return decimalFormat.format(value);

      }



}

File: app/src/main/java/com/sofrosyn/a6gnlsolarcalculator/logic/Calculator.kt
package com.sofrosyn.a6gnlsolarcalculator.logic

import java.text.DecimalFormat

object Calculator {
    private var inverterRatingKva = 0.0
    private const val backUpTime = 0
    private const val timeToCharge = 0
    private var minRatingChargeController = 0
    private var noSolarPanels = 0
    private const val LOADFACTOR = 0.8
    private const val AMPHOURBATTERY = 200
    private const val HOURS = 8



    fun loadInKva(load: Double): Double {
        inverterRatingKva = load / LOADFACTOR / 1000
        return inverterRatingKva
    }

    fun batteryBackupTime(load: Float, inverterVoltage: Float, batteryAmp: Float): Float {
        val cal = load / inverterVoltage
        return batteryAmp / cal
    }

    fun timeToCharge(batteryAmps: Float, inverterVoltage: Float): Float {
        return batteryAmps / inverterVoltage
    }

    fun minCurrentRatingController(panelPowerRating: Int, panelVoltage: Int, noSolarPanel: Int): Int {
        minRatingChargeController = panelPowerRating * noSolarPanel
        return minRatingChargeController / panelVoltage
    }

    fun totalPanels(inverterVoltage: Int, panelRating: Int): Int {
        val a = AMPHOURBATTERY * inverterVoltage
        val b = HOURS * panelRating
        noSolarPanels = a / b
        return noSolarPanels
    }

    fun ampsConverter(batterNumber: Int, divisor: Int, ampHourBattery: Int): Int {
        val base = batterNumber / divisor
        return base * ampHourBattery
    }

    fun KvaMargin(load: Double): Double {
        return load + 0.3 * load
    }

    fun formatDecimal(value: Double): String {
        val decimalFormat = DecimalFormat("#.##")
        return decimalFormat.format(value)
    }

    }

